### PR TITLE
Adds a -no_minify CL argument to KMW's build.sh.

### PR DESCRIPTION
**Pull request by jahorton**
This new argument bypasses the use of Closure for minification and thus allows our Developer build infrastructure a path to avoid a Java requirement.  Developed in refererence/as an enhancement to #540's work.

It also allows faster compilation when doing testing with unminified pages and polishes up build.sh a bit.

It also now results in publishing both a minified and unminified version of KeymanWeb officially within the release section - unminified versions are published under the `/release/unminified` subfolder.

### CODE REVIEW

**Comment by jahorton** [comment]
Well, the build products to be copied _are_ JavaScript, so I think that comment should stay as is.  I'll add things to `readme.md` though.

File: web/README.md [hunk 1]
```diff
@@ -21,8 +21,25 @@ The following folders contain the distribution for KeymanWeb:
...
+
 To view pages using compiled KeymanWeb,
 1. cd to **keyman/web/source**
 2. Run `/.build.sh`
 
-Refer to the samples for usage details.
+### `build.sh` options:
+
+*  `-ui`
+    - to compile desktop user interface modules to output folder
```
Comment (jahorton) [inline web/README.md]: Best viewed [in code mode](https://github.com/keymanapp/keyman/tree/web-no-minify-build-option/web).

File: web/source/build.sh [hunk 2]
```diff
@@ -222,13 +228,13 @@ if [ $BUILD_EMBED = true ]; then
...
-    minify keyman.js $EMBED_OUTPUT SIMPLE_OPTIMIZATIONS "KeymanBase.__BUILD__=$BUILD"
-    assert $EMBED_OUTPUT/keyman.js 
+    echo Embedded TypeScript compiled as $INTERMEDIATE/keyman.js
 
-    echo Compiled embedded application saved as $EMBED_OUTPUT/keyman.js
+    if [ $DO_MINIFY = true ]; then
+        minify keyman.js $EMBED_OUTPUT SIMPLE_OPTIMIZATIONS "KeymanBase.__BUILD__=$BUILD"
+        assert $EMBED_OUTPUT/keyman.js
+        echo Compiled embedded application saved as $EMBED_OUTPUT/keyman.js
+    fi
```
Comment (darcywong00) [inline web/source/build.sh]: When I build `./build.sh -no_minify`, I'm not seeing a current `/web/release/embedded/keyman.js`. Shouldn't the build script copy `keyman.js` to `$EMBED_OUTPUT/`? 

And for keymanweb.js?

The `asserts` probably should be done after the `fi` statements

**Comment by jahorton** [comment]
In that case, the "intermediate" (unminified) build products are updated but not the "release" artifacts; this is as presently intended.  This is why the asserts are placed as-is, as well.

From the current `README.md`:

```
intermediate    	Intermediate compiled code; TypeScript source is compiled here before it is minified.
release/web		Fully-compiled KeymanWeb modules for testing
release/embedded	Fully-compiled KMEA/KMEI modules for inclusion in mobile app builds
```

**Comment by darcywong00** [comment]
I see that distinction. My concern is that KMEA and KMEI are expecting `/web/release/embedded/keyman.js` and now those could be stale or missing.

I could see a case where a a mobile app developer would still like to use unminified KMW source

**Comment by jahorton** [comment]
I feel like keeping the build project folders separate makes things more clear, though.  You know exactly what you're getting depending upon the folder you draw from, rather than having one result suddenly change behavior under select conditions.  I might be treating build.sh a bit too much like a program, though.

**Comment by jahorton** [comment]
Talking with @mcdurdin - we'll opt for the following:

```
intermediate    	     Intermediate compiled code; TypeScript source is compiled here before it is minified.
release/web		     Fully-compiled KeymanWeb modules
release/embedded	     Fully-compiled KMEA/KMEI modules for inclusion in mobile app builds
release/unminified/web       Fully-compiled but non-minified KeymanWeb modules
release/unminified/embedded  Fully-compiled KMEA/KMEI modules for inclusion in mobile app builds
```

We want to keep the minified v unminified folders distinct and separate, but it's good to have a decoupling here in case we ever need to incorporate extra build stages after the TypeScript compilation.

**Comment by mcdurdin** [comment]
NOTE:  Before merging, the TC script for our downloads.keyman.com *.zip construction needs a rework to include the unminified version of 'web' in that .zip.  'embed' products should still be left out of the archive.

File: web/README.md [hunk 3]
```diff
@@ -5,24 +5,43 @@ The Original Code is (C) 2017 SIL International
...
+*  `-test`
+    - to compile for testing without copying resources or updating the saved version number.
+*  `-embed`
+    - to compile only the KMEA/KMEI embedded engine.
+*  `-web`
+    - to compile only the KeymanWeb engine.
+*  `-debug_embedded`
+    - to compile a readable version of the embedded KMEA/KMEI code
+*  `-no_minify`
+    - to disable the minification '/release/' build sections
```
Comment (darcywong00) [inline web/README.md]: Can you clarify the `-no_minify` option with the new directory structure `/release/unminified`?

Comment (darcywong00) [reply web/README.md]: and add the new `-clean` option too

File: web/source/build.sh [hunk 4]
```diff
@@ -4,21 +4,26 @@
 #
 
 display_usage ( ) {
-    echo "build.sh [-ui | -test | -embed | -web | -debug_embedded]"
+    echo "build.sh [-ui | -test | -embed | -web | -debug_embedded] [-no_minify]"
```
Comment (darcywong00) [inline web/source/build.sh]: sorry, one more niddle

`| [-clean]`

**Pull request merged**